### PR TITLE
To add brief intro for CPU backend optimization

### PR DESCRIPTION
**Pull request by ZailiWang**
Previous #103666 was reverted as the new doc was not included in toctree. Fixed this and created this new PR.

cc @svekars @carljparker @frank-wei @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Comment by ZailiWang** [comment]
The doc test still fails due the warning
```
/opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/__init__.py:docstring of torch.compile:1: WARNING: duplicate object description of torch.compile, other instance in compile/generated/torch.compile, use :noindex: for one of them
```
but I've no idea where does this come from.

**Review by jgong5 — APPROVED** [review]

**Comment by ZailiWang** [comment]
Hi @svekars would you help check why this PR repeatedly fails doc test? The WARNING msg is pasted above, but it seems nothing to do with my changes. Thanks a lot.

**Comment by svekars** [comment]
> Hi @svekars would you help check why this PR repeatedly fails doc test? The WARNING msg is pasted above, but it seems nothing to do with my changes. Thanks a lot.

I believe this was fixed here: https://github.com/pytorch/pytorch/pull/104911. Waiting for the build to complete to verify.

**Review by svekars — COMMENTED** [review]
A fe minor editorial suggestions. Thanks!

**Comment by svekars** [comment]
@ZailiWang can you please rebase and resolve the merge conflict?

File: docs/source/compile/best-practice-for-backends.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Best Practice for Backends
+==========================
+
+x86 CPU
+-----------------
+
+Compiled workloads on modern x86 CPUs are usually optimized by SIMD instruction sets. SIMD (Single Instruction Multiple Data) is a typical parallel processing technique for high performance computing, such as deep learning model training and inference. With SIMD applied, each compute unit performs the same instruction with different allocated data at any given timeslot. The most commonly deployed x86 instruction set architectures (ISAs) enabling SIMD include `AVX, AVX2, AVX-512 <https://en.wikipedia.org/wiki/Advanced_Vector_Extensions>` and `AMX <https://en.wikipedia.org/wiki/Advanced_Matrix_Extensions>`.
+
+Which ISAs are supported by a machine can be checked with `collect_env script <https://github.com/pytorch/pytorch/blob/main/torch/utils/collect_env.py>`. As the script provides complete environment information for PyTorch, we can use ``grep`` to extract the line containing ISA information.
```
[the remark this answers is not in the document]
Comment (ZailiWang) [reply docs/source/compile/best-practice-for-backends.rst:9]: There is a spell out of ISA above, at the last sentence of 1st paragraph.

File: docs/source/compile/best-practice-for-backends.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
+Best Practice for Backends
```
[the remark this answers is not in the document]
Comment (ZailiWang) [reply docs/source/compile/best-practice-for-backends.rst]: corrected this along with filename correction.

**Pull request closed without merging**